includes/api/class-wc-admin-rest-products-controller.php

```diff
@@ -87,30 +87,51 @@ protected function prepare_objects_query( $request ) {
 	 * @return WP_Error|WP_REST_Response
 	 */
 	public function get_items( $request ) {
-		add_filter( 'posts_where', array( __CLASS__, 'add_wp_query_post_title_filter' ), 10, 2 );
+		add_filter( 'posts_where', array( __CLASS__, 'add_wp_query_product_search_filter' ), 10, 2 );
+		add_filter( 'posts_join', array( __CLASS__, 'add_wp_query_product_search_join' ), 10, 2 );
 		$response = parent::get_items( $request );
-		remove_filter( 'posts_where', array( __CLASS__, 'add_wp_query_post_title_filter' ), 10 );
+		remove_filter( 'posts_where', array( __CLASS__, 'add_wp_query_product_search_filter' ), 10 );
+		remove_filter( 'posts_join', array( __CLASS__, 'add_wp_query_product_search_join' ), 10 );
 		return $response;
 	}
 
 	/**
-	 * Add post title searching to WP Query
+	 * Allow searching by product name or sku in WP Query
 	 *
 	 * @param string $where Where clause used to search posts.
 	 * @param object $wp_query WP_Query object.
 	 * @return string
 	 */
-	public static function add_wp_query_post_title_filter( $where, $wp_query ) {
+	public static function add_wp_query_product_search_filter( $where, $wp_query ) {
 		global $wpdb;
 
-		$post_title_search = $wp_query->get( 'post_title__like' );
-		if ( $post_title_search ) {
-			$post_title_search = $wpdb->esc_like( $post_title_search );
-			$post_title_search = ' \'%' . $post_title_search . '%\'';
-			$where            .= ' AND ' . $wpdb->posts . '.post_title LIKE ' . $post_title_search;
+		$product_search = $wp_query->get( 'product_search' );
+		if ( $product_search ) {
+			$product_search = $wpdb->esc_like( $product_search );
+			$product_search = ' \'%' . $product_search . '%\'';
+			$where         .= ' AND (' . $wpdb->posts . '.post_title LIKE ' . $product_search . ' OR ';
+			$where         .= 'ps_post_meta.meta_key = "_sku" AND ps_post_meta.meta_value LIKE ' . $product_search . ')';
```

Comment:
Any performance concerns here with the meta query?

Reply:
It's certainly not as quick as just searching the posts table, but also not terrible performance and uses the same basic meta query SQL syntax that core does.

if we want to offer SKU searching, we don't have a lot of options here.